### PR TITLE
feat(docs): Refresh story writing meta docs

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Updates our story writing guide to reflect current practices, but not overly prescriptive in terms of structure.  This will be updated again once Yassine and I have piloted the standard template.

Includes:
- How to use vscode snippets for quickly setting things up.
- Example MDX blocks and descriptions of what meta-information we'd like in each file.
- Some small notes about args / controls and how to use them on granular stories. 

* Refreshes our story writing guide to include current practices of MDX and Controls. 

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.15.3-alpha.0beb69.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600766e50c96923830620d44--gamut-preview.netlify.app>https://600766e50c96923830620d44--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600766e50c96923830620d44>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.15.8-alpha.70f543.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b0a80d658dd13484de1c6--gamut-preview.netlify.app>https://600b0a80d658dd13484de1c6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b0a80d658dd13484de1c6>Deploy Logs</a></p>

File: packages/styleguide/stories/Meta/Stories.stories.mdx [hunk 1]
```diff
@@ -4,60 +4,85 @@ import { Meta } from '@storybook/addon-docs/dist/blocks';
...
+    subtitle: `We're still building these opinions out, so don't expect all stories to adhere just yet!
 `,
   }}
 />
 
-## Story Structure
+## Quick Start
 
-In our fledgling collective opinion, a good component story page consists of:
+We've provided a few helpful vscode snippets to help you get through boilerplate. To use these simply start to type these strings in your editor and pick the template and fill out the tab targets.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Meta/Stories.stories.mdx:14]: Nothing is ever simple 😄 

```suggestion
We've provided a few helpful vscode snippets to help you get through boilerplate. To use these start to type these strings in your editor and pick the template and fill out the tab targets.
```

File: packages/styleguide/stories/Meta/Stories.stories.mdx [hunk 2]
```diff
@@ -4,60 +4,85 @@ import { Meta } from '@storybook/addon-docs/dist/blocks';
...
+When you make a new story theres a few things to keep in mind:
 
-Give this story the same name as its component in camelCase.
-Its description should describe at least:
+- The folder structure is indicative of our flavor of atomic design.
+- The folder struture is identical to the storybook heirarchy generated.
 
-- What the component does
-- What it would typically be used for
+First find the right folder for your story in `packages/styleguide/stories` (e.g. `Atoms` | `Molecules` | `Organisms`).Once you've found it create a new file `#{COMPONENT_NAME}.stories.mdx` (if you need an adjacent file to store variables create a folder of the component name and create an `index.stories.mdx` inside of it).
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Meta/Stories.stories.mdx:27]:
```suggestion
First find the right folder for your story in `packages/styleguide/stories` (e.g. `Atoms` | `Molecules` | `Organisms`).
Once you've found it create a new file `#{COMPONENT_NAME}.stories.mdx` (if you need an adjacent file to store variables create a folder of the component name and create an `index.stories.mdx` inside of it).
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great! Love how you simplified & distilled this down to the essentials. ❤️

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.15.8-alpha.ff57e5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b10c81f098b1e0a567cf2--gamut-preview.netlify.app>https://600b10c81f098b1e0a567cf2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b10c81f098b1e0a567cf2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.16.2-alpha.bfa466.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b13c52698e4291ec80a80--gamut-preview.netlify.app>https://600b13c52698e4291ec80a80--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b13c52698e4291ec80a80>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b3c8011748a221c75348f--gamut-preview.netlify.app>https://600b3c8011748a221c75348f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b3c8011748a221c75348f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.17.1-alpha.a2997c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.13.0-alpha.ef9901.0<br/>@codecademy/styleguide@21.18.0-alpha.ef9901.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b3ea80590d2284b1bc5ed--gamut-preview.netlify.app>https://600b3ea80590d2284b1bc5ed--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b3ea80590d2284b1bc5ed>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.18.1-alpha.32efe4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b56245dcdea4ce2953711--gamut-preview.netlify.app>https://600b56245dcdea4ce2953711--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b56245dcdea4ce2953711>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.18.2-alpha.d321c8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b6112e75f9a5922a88723--gamut-preview.netlify.app>https://600b6112e75f9a5922a88723--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b6112e75f9a5922a88723>Deploy Logs</a></p>

**Pull request merged**